control-plane/grpc/proto/v1/volume/volume.proto

```diff
@@ -419,11 +419,13 @@ message DeleteSnapshotReply {
 message GetSnapshotsRequest {
   // filter snapshots
   oneof filter {
-    common.VolumeFilter           volume = 1;
-    common.VolumeSnapshotFilter snapshot = 2;
+    common.VolumeFilter                  volume = 1;
+    common.VolumeSnapshotFilter volume_snapshot = 2;
+    common.SnapshotFilter              snapshot = 3;
   }
-  // ignore 404 not found errors
-  bool                   ignore_notfound = 3;
+  // ignore 404 not found errors.
+  // Left few field numbers in between to accomodate any future filters.
+  bool                   ignore_notfound = 6;
```

Comment:
Can we do that? I thought we cannot add things in between. It always needs to be a new one.

Reply:
The textual order is apparently largely irrelevant. Quoting from protobuf doc:

> Field numbers may be declared in any order in a .proto file. The order chosen has no effect on how the messages are serialized.

Reply:
No particular reason apart from some buffer indices keeping in mind future ordering visibility. Let me know what's preferable.

```
message GetSnapshotsRequest {
  // filter snapshots
  oneof filter {
    common.VolumeFilter                  volume = 1;
    common.VolumeSnapshotFilter volume_snapshot = 2;
    common.SnapshotFilter              snapshot = 3;
    <some new filter>                 newfilter1 = 4;
   <some new filter>                 newfilter2 = 5;
  }
  bool                   ignore_notfound = 6;
}
```

 vs 

```
message GetSnapshotsRequest {
  // filter snapshots
  oneof filter {
    common.VolumeFilter                  volume = 1;
    common.VolumeSnapshotFilter volume_snapshot = 2;
    common.SnapshotFilter              snapshot = 3;
    <some new filter>                    newfilter1 = 5;
   <some new filter>                  newfilter2 = 6;
  }
  bool                   ignore_notfound = 4;
}
```

Reply:
hmm I think second is fine since we might not even add more filters.